packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,435 @@
+   * The behavior is intended to be used with a single selection style list, as multi selection style list should not perform disclosures.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * An indicator to be displayed when no children are yet present.
+   */
+  initialLoadingIndicator: PropTypes.element,
+  /**
+   * Whether or not the child list items should have a border color applied.
+   */
+  isDivided: PropTypes.bool,
+  /**
+   * Determines whether or not the loading indicator is visible and if callbacks are triggered.
+   */
+  isFinishedLoading: PropTypes.bool,
+  /**
+   * Whether or not the list is selectable.
+   */
+  isSelectable: PropTypes.bool,
+  /**
+   * A callback event that will be triggered when selection state changes.
+   */
+  onChange: PropTypes.func,
+  /**
+   * Callback trigger when new list items are requested.
+   */
+  onRequestItems: PropTypes.func,
+  /**
+   * An indicator to be displayed at the end of the current loaded children.
+   */
+  progressiveLoadingIndicator: PropTypes.element,
+  /**
+   * An array of the currectly selected indexes.
+   */
+  selectedIndexes: PropTypes.array,
+};
+
+const defaultProps = {
+  children: [],
+  disableUnselectedItems: false,
+  hasChevrons: false,
+  isDivided: false,
+  isFinishedLoading: false,
+  isSelectable: false,
+  selectedIndexes: [],
+};
+
+const createSpacer = (height, index) => (
+  <List.Item
+    isSelectable={false}
+    className={cx(['spacer'])}
+    style={{ height }}
+    key={`infinite-spacer-${index}`}
+  />
+);
+
+class InfiniteList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.update = this.update.bind(this);
+    this.enableListeners = this.enableListeners.bind(this);
+    this.disableListeners = this.disableListeners.bind(this);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.updateItemCache = this.updateItemCache.bind(this);
+    this.initializeItemCache = this.initializeItemCache.bind(this);
+    this.updateScrollGroups = this.updateScrollGroups.bind(this);
+    this.handleRenderCompletion = this.handleRenderCompletion.bind(this);
+    this.handleResize = this.resizeDebounce(this.handleResize.bind(this));
+    this.resetTimeout = this.resetTimeout.bind(this);
+    this.wrapChild = this.wrapChild.bind(this);
+
+    this.initializeItemCache(props);
+  }
+
+  componentDidMount() {
+    if (!this.listenersAdded) {
+      this.enableListeners();
+    }
+    this.handleRenderCompletion();
+    this.updateScrollGroups();
+    this.update(null, false);
+  }
+
+  componentWillReceiveProps(newProps) {
+    const newChildCount = React.Children.count(newProps.children);
+    if (newChildCount > this.childCount) {
+      this.lastChildIndex = this.childCount;
+      this.loadingIndex += 1;
+      this.updateItemCache(newProps);
+    } else if (newChildCount < this.childCount) {
+      this.initializeItemCache(newProps);
+    }
+  }
+
+  componentDidUpdate() {
+    if (!this.listenersAdded) {
```

Comment:
is there potential in loosing the listeners after mount?

Reply:
It shouldn't, though I could see adding a key or the list to ensure it.